### DISCUSSION TITLE
ktx create or basisu ?

### DISCUSSION THREAD

**Discussion by michaeldll** [KhronosGroup/KTX-Software#866] [Q&A] [answered]
Hello everyone. I'm a WebGL developer, currently developing custom tooling for my agency. 
I'm wondering about current best practices concerning compressing PNG files into supercompressed UASTC or ETC1S textures: should I base my CLI tooling on `ktx create` or `basisu` ? As I understand it, `ktx create` is the latest tooling, following `basisu` and `toktx`'s release dates.
Thanks and hope I'm in the right place.

**Comment by MarkCallow** [comment, marked as answer]
`ktx create`, `basisu` and `toktx` all use the same encoder which originates with `basisu` and is incorporated in `libktx`. You should use whichever tool has the functionality you need but be aware that `toktx` is deprecated and will be removed in the near future.

**Reply by michaeldll** [reply]
Thank you.
